crates/metadata/src/lib.rs

```diff
@@ -28,6 +28,8 @@ pub mod layout;
 mod specs;
 mod utils;
 
+pub use ink_primitives::LangError;
```

Comment:
> So this reexport is for the situation where a client has a dep on ink_metadata (and not ink which already exports it via ink::LangError) and just wants to easily grab the LangError type?

Yep.

I wasn't able to have the type directly in the `ink_metadata` crate because we only compile it for `std`, and we need to use this type as part of the codegen (which is `no_std`).